src/test/java/org/prebid/server/auction/PrivacyEnforcementServiceTest.java

```diff
@@ -0,0 +1,662 @@
+
+        final BidRequest bidRequest = givenBidRequest(givenSingleImp(
+                singletonMap(BIDDER_NAME, 1)),
+                bidRequestBuilder -> bidRequestBuilder
+                        .user(user)
+                        .device(device)
+                        .regs(regs));
+
+        // when
+        final Future<Map<String, PrivacyEnforcementResult>> masked = privacyEnforcementService
+                .mask(bidderToUser, extUser, null, singletonList(BIDDER_NAME), emptyMap(), bidRequest, PUBLISHER_ID, timeout);
+
+        // then
+        verify(gdprService).isGdprEnforced(isNull(), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout));
+        verifyNoMoreInteractions(gdprService);
+        verify(metrics).updateGdprMaskedMetric(eq(BIDDER_NAME));
+        verifyNoMoreInteractions(metrics);
+
+        final Regs expectedRegs = Regs.of(1, mapper.valueToTree(ExtRegs.of(1)));
+        //Coppa includes all masked fields for Gdpr
+        final Device expectedDevice = givenCoppaMaskedDevice(deviceBuilder -> deviceBuilder.lmt(1));
+        final PrivacyEnforcementResult expected = PrivacyEnforcementResult.of(userCoppaMasked(), expectedDevice,
+                expectedRegs);
+        assertThat(masked.succeeded()).isTrue();
+        assertThat(masked.result()).hasSize(1)
+                .containsOnly(entry(BIDDER_NAME, expected));
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldReturnFailedFutureWhenGdprServiceIsReturnFailedFuture() {
+        // given
+        given(gdprService.resultByVendor(any(), any(), any(), any(), any()))
+                .willReturn(
+                        Future.failedFuture(new InvalidRequestException(
+                                "Error when retrieving allowed purpose ids in a reason of invalid consent string")),
+                        Future.failedFuture(new InvalidRequestException(
+                                "Error when checking if vendor is allowed in a reason of invalid consent string")));
+
+        final ExtUser extUser = ExtUser.builder().build();
+        final User user = notMaskedUser();
+        final Device device = givenNotMaskedDevice(deviceBuilder -> deviceBuilder.lmt(1));
+        final Regs regs = Regs.of(1, null);
+        final Map<String, User> bidderToUser = singletonMap(BIDDER_NAME, user);
+
+        final BidRequest bidRequest = givenBidRequest(givenSingleImp(
+                singletonMap(BIDDER_NAME, 1)),
+                bidRequestBuilder -> bidRequestBuilder
+                        .user(user)
+                        .device(device)
+                        .regs(regs));
+
+        // when
+        final Future<Map<String, PrivacyEnforcementResult>> firstFuture = privacyEnforcementService
+                .mask(bidderToUser, extUser, null, singletonList(BIDDER_NAME), emptyMap(), bidRequest, PUBLISHER_ID, timeout);
+        final Future<Map<String, PrivacyEnforcementResult>> secondFuture = privacyEnforcementService
```

Comment:
I just wanted to include all failed futures, and show them in one test. Also, we can see all messages that may occur. 
it would be more practical if we had different Exception types. 

If you prefer, I can remove one.